Scanner crashes (nil deref)

## Description

Scanner crashes.

## Expected Behavior

Scanner does not crash.

## Actual Behavior

```
time="2024-10-14T16:52:18Z" level=info msg=Running... app=openclarity
time="2024-10-14T16:52:18Z" level=info msg="Initializing configuration..." app=openclarity
time="2024-10-14T16:52:18Z" level=info msg="Using config file (/etc/openclarity/config.yaml):\nexploits:\n  enabled: true\n  input_from_vuln: true\n  inputs:\n  - input: /mnt/snapshot/image.tar\n    input_type: oci-archive\n    strip_path_from_result: true\n  scanners_config:\n    exploit_db:\n      base_url: http://openclarity-exploit-db-server:1326\n  scanners_list:\n  - exploitdb\ninfofinder:\n  enabled: true\n  inputs:\n  - input: /mnt/snapshot/image.tar\n    input_type: oci-archive\n    strip_path_from_result: true\n  scanners_config:\n    ssh_topology: {}\n  scanners_list:\n  - sshTopology\n  strip_input_paths: false\nmalware:\n  enabled: false\n  inputs: ]\n  scanners_config:\n    clam:\n      alternative_freshclam_mirror_url: \"\"\n      clam_daemon_binary_path: \"\"\n      clam_daemon_client_binary_path: \"\"\n      clam_daemon_config_path: \"\"\n      clamscan_binary_path: \"\"\n      clamscan_exclude_dirs: ]\n      clamscan_exclude_files: ]\n      freshclam_binary_path: \"\"\n      freshclam_config_path: \"\"\n      use_native_clamscan: false\n    yara:\n      cache_dir: \"\"\n      compiled_rule_url: \"\"\n      directories_to_scan: ]\n      rule_sources: ]\n      yara_binary_path: \"\"\n      yarac_binary_path: \"\"\n  scanners_list: ]\n  strip_input_paths: false\nmisconfiguration:\n  enabled: true\n  inputs:\n  - input: /mnt/snapshot/image.tar\n    input_type: oci-archive\n    strip_path_from_result: true\n  scanners_config:\n    cisdocker:\n      registry: null\n      timeout: 0\n    lynis:\n      binary_path: \"\"\n  scanners_list:\n  - lynis\n  - cisdocker\n  strip_input_paths: false\nplugins:\n  binary_artifacts_clean: false\n  binary_artifacts_path: \"\"\n  binary_mode: false\n  enabled: false\n  inputs: ]\n  scanners_config: null\n  scanners_list: ]\nrootkits:\n  enabled: false\n  inputs: ]\n  scanners_config:\n    chkrootkit:\n      binary_path: \"\"\n  scanners_list: ]\n  strip_input_paths: false\nsbom:\n  analyzers_config:\n    syft:\n      exclude_paths: ]\n      local_image_scan: false\n      registry: null\n      scope: squashed\n    trivy:\n      cache_dir: \"\"\n      local_image_scan: false\n      registry: null\n      temp_dir: \"\"\n      timeout: 300\n  analyzers_list:\n  - syft\n  - trivy\n  - windows\n  - gomod\n  enabled: true\n  inputs:\n  - input: /mnt/snapshot/image.tar\n    input_type: oci-archive\n    strip_path_from_result: null\n  local_image_scan: false\n  merge_with: ]\n  output_format: cyclonedx-json\n  registry:\n    auths: ]\n    skip-verify-tls: false\n    use-http: false\nsecrets:\n  enabled: true\n  inputs:\n  - input: /mnt/snapshot/image.tar\n    input_type: oci-archive\n    strip_path_from_result: true\n  scanners_config:\n    gitleaks:\n      binary_path: \"\"\n  scanners_list:\n  - gitleaks\n  strip_input_paths: false\nvulnerabilities:\n  enabled: true\n  input_from_sbom: true\n  inputs: ]\n  local_image_scan: false\n  registry: null\n  scanners_config:\n    grype:\n      local_grype_config:\n        db_root_dir: \"\"\n        listing_file_timeout: 0\n        listing_url: \"\"\n        local_image_scan: false\n        max_allowed_built_age: 0\n        registry: null\n        scope: \"\"\n        update_db: false\n        update_timeout: 0\n      mode: REMOTE\n      remote_grype_config:\n        grype_server_address: openclarity-grype-server:9991\n        grype_server_schemes: ]\n        grype_server_timeout: 120000000000\n    trivy:\n      cache_dir: \"\"\n      registry: null\n      server_addr: http://openclarity-trivy-server:9992\n      server_token: \"\"\n      temp_dir: \"\"\n      timeout: 300\n  scanners_list:\n  - grype\n  - trivy\n" app=openclarity
time="2024-10-14T16:52:33Z" level=info msg="Running scanners..." app=openclarity
time="2024-10-14T16:52:33Z" level=info msg="Running family \"misconfiguration\" in progress..." app=openclarity family=misconfiguration
time="2024-10-14T16:52:33Z" level=info msg="Scanning inputs in progress..." inputs="[oci-archive:/mnt/snapshot/image.tar]" app=openclarity family=misconfiguration
time="2024-10-14T16:52:33Z" level=info msg="Scan job \"lynis\" for input oci-archive:/mnt/snapshot/image.tar in progress..." scanner=lynis input="oci-archive:/mnt/snapshot/image.tar" app=openclarity family=misconfiguration
time="2024-10-14T16:52:33Z" level=info msg="Scan job \"cisdocker\" for input oci-archive:/mnt/snapshot/image.tar in progress..." app=openclarity family=misconfiguration scanner=cisdocker input="oci-archive:/mnt/snapshot/image.tar"
time="2024-10-14T16:52:33Z" level=warning msg="Scan job \"cisdocker\" for input oci-archive:/mnt/snapshot/image.tar failed" app=openclarity family=misconfiguration scanner=cisdocker input="oci-archive:/mnt/snapshot/image.tar" error="unsupported source type=oci-archive"
time="2024-10-14T16:52:33Z" level=warning msg="Scan job \"lynis\" for input oci-archive:/mnt/snapshot/image.tar failed" error="failed to convert input to filesystem: failed to expand container to rootfs directory: unable to get image: unable to parse image from src oci-archive:/mnt/snapshot/image.tar: unable to detect input for '/mnt/snapshot/image.tar', errs: unable to parse OCI directory as an image: unexpected media type for sha256:beee3efa712609c45c5a5822bc633fbf36cb7917cd43108d4edcad8a9f7c1d97: application/vnd.oci.image.index.v1+json" family=misconfiguration scanner=lynis input="oci-archive:/mnt/snapshot/image.tar" app=openclarity
time="2024-10-14T16:52:33Z" level=error msg="Scanning inputs failed with 1 errors" error="scan job \"cisdocker\" failed for input oci-archive:/mnt/snapshot/image.tar, reason: unsupported source type=oci-archive\nscan job \"lynis\" failed for input oci-archive:/mnt/snapshot/image.tar, reason: failed to convert input to filesystem: failed to expand container to rootfs directory: unable to get image: unable to parse image from src oci-archive:/mnt/snapshot/image.tar: unable to detect input for '/mnt/snapshot/image.tar', errs: unable to parse OCI directory as an image: unexpected media type for sha256:beee3efa712609c45c5a5822bc633fbf36cb7917cd43108d4edcad8a9f7c1d97: application/vnd.oci.image.index.v1+json" app=openclarity family=misconfiguration
time="2024-10-14T16:52:33Z" level=error msg="Family \"misconfiguration\" finished with error" app=openclarity family=misconfiguration error="failed to process inputs for misconfigurations: scan job \"cisdocker\" failed for input oci-archive:/mnt/snapshot/image.tar, reason: unsupported source type=oci-archive\nscan job \"lynis\" failed for input oci-archive:/mnt/snapshot/image.tar, reason: failed to convert input to filesystem: failed to expand container to rootfs directory: unable to get image: unable to parse image from src oci-archive:/mnt/snapshot/image.tar: unable to detect input for '/mnt/snapshot/image.tar', errs: unable to parse OCI directory as an image: unexpected media type for sha256:beee3efa712609c45c5a5822bc633fbf36cb7917cd43108d4edcad8a9f7c1d97: application/vnd.oci.image.index.v1+json"
time="2024-10-14T16:52:33Z" level=info msg="Running family \"sbom\" in progress..." family=sbom app=openclarity
time="2024-10-14T16:52:34Z" level=info msg="Scanning inputs in progress..." family=sbom app=openclarity inputs="[oci-archive:/mnt/snapshot/image.tar]"
time="2024-10-14T16:52:34Z" level=info msg="Generating hash for input /mnt/snapshot/image.tar" family=sbom app=openclarity
time="2024-10-14T16:52:34Z" level=info msg="Skip generating hash in the case of image"
time="2024-10-14T16:52:34Z" level=info msg="Scan job \"syft\" for input oci-archive:/mnt/snapshot/image.tar in progress..." app=openclarity family=sbom scanner=syft input="oci-archive:/mnt/snapshot/image.tar"
time="2024-10-14T16:52:34Z" level=info msg="Scan job \"trivy\" for input oci-archive:/mnt/snapshot/image.tar in progress..." family=sbom input="oci-archive:/mnt/snapshot/image.tar" scanner=trivy app=openclarity
time="2024-10-14T16:52:34Z" level=info msg="Scan job \"windows\" for input oci-archive:/mnt/snapshot/image.tar in progress..." app=openclarity family=sbom input="oci-archive:/mnt/snapshot/image.tar" scanner=windows
time="2024-10-14T16:52:34Z" level=info msg="Scan job \"gomod\" for input oci-archive:/mnt/snapshot/image.tar in progress..." input="oci-archive:/mnt/snapshot/image.tar" app=openclarity family=sbom scanner=gomod
time="2024-10-14T16:52:34Z" level=warning msg="Scan job \"windows\" for input oci-archive:/mnt/snapshot/image.tar failed" family=sbom input="oci-archive:/mnt/snapshot/image.tar" error="skipping analyzing unsupported source type: oci-archive" scanner=windows app=openclarity
time="2024-10-14T16:52:34Z" level=warning msg="Scan job \"gomod\" for input oci-archive:/mnt/snapshot/image.tar failed" input="oci-archive:/mnt/snapshot/image.tar" app=openclarity error="unsupported input type=oci-archive" family=sbom scanner=gomod
time="2024-10-14T16:52:34Z" level=warning msg="Scan job \"syft\" for input oci-archive:/mnt/snapshot/image.tar failed" input="oci-archive:/mnt/snapshot/image.tar" error="failed to create source analyzer=syft: an error occurred attempting to resolve '/mnt/snapshot/image.tar': oci-archive: unable to parse OCI directory as an image: unexpected media type for sha256:beee3efa712609c45c5a5822bc633fbf36cb7917cd43108d4edcad8a9f7c1d97: application/vnd.oci.image.index.v1+json" app=openclarity family=sbom scanner=syft
2024/10/14 16:52:35 INFO Java DB Repository repository=ghcr.io/aquasecurity/trivy-db:2
2024/10/14 16:52:35 INFO Downloading the Java DB...
time="2024-10-14T16:52:35Z" level=warning msg="Scan job \"trivy\" for input oci-archive:/mnt/snapshot/image.tar failed" family=sbom input="oci-archive:/mnt/snapshot/image.tar" error="failed to generate SBOM: image scan error: scan error: scan failed: failed analysis: analyze error: pipeline error: failed to analyze layer (sha256:bff6683ea42c9216ad9427b6487fd00feadce5219a3a26e2923d563f6c4c865e): post analysis error: post analysis error: Unable to initialize the Java DB: Java DB update failed: DB download error: unacceptable media type: application/vnd.aquasec.trivy.db.layer.v1.tar+gzip" scanner=trivy app=openclarity
time="2024-10-14T16:52:35Z" level=error msg="Scanning inputs failed with 1 errors" error="scan job \"windows\" failed for input oci-archive:/mnt/snapshot/image.tar, reason: skipping analyzing unsupported source type: oci-archive\nscan job \"gomod\" failed for input oci-archive:/mnt/snapshot/image.tar, reason: unsupported input type=oci-archive\nscan job \"syft\" failed for input oci-archive:/mnt/snapshot/image.tar, reason: failed to create source analyzer=syft: an error occurred attempting to resolve '/mnt/snapshot/image.tar': oci-archive: unable to parse OCI directory as an image: unexpected media type for sha256:beee3efa712609c45c5a5822bc633fbf36cb7917cd43108d4edcad8a9f7c1d97: application/vnd.oci.image.index.v1+json\nscan job \"trivy\" failed for input oci-archive:/mnt/snapshot/image.tar, reason: failed to generate SBOM: image scan error: scan error: scan failed: failed analysis: analyze error: pipeline error: failed to analyze layer (sha256:bff6683ea42c9216ad9427b6487fd00feadce5219a3a26e2923d563f6c4c865e): post analysis error: post analysis error: Unable to initialize the Java DB: Java DB update failed: DB download error: unacceptable media type: application/vnd.aquasec.trivy.db.layer.v1.tar+gzip" family=sbom app=openclarity
time="2024-10-14T16:52:35Z" level=error msg="Family \"sbom\" finished with error" app=openclarity error="failed to process inputs for sbom: scan job \"windows\" failed for input oci-archive:/mnt/snapshot/image.tar, reason: skipping analyzing unsupported source type: oci-archive\nscan job \"gomod\" failed for input oci-archive:/mnt/snapshot/image.tar, reason: unsupported input type=oci-archive\nscan job \"syft\" failed for input oci-archive:/mnt/snapshot/image.tar, reason: failed to create source analyzer=syft: an error occurred attempting to resolve '/mnt/snapshot/image.tar': oci-archive: unable to parse OCI directory as an image: unexpected media type for sha256:beee3efa712609c45c5a5822bc633fbf36cb7917cd43108d4edcad8a9f7c1d97: application/vnd.oci.image.index.v1+json\nscan job \"trivy\" failed for input oci-archive:/mnt/snapshot/image.tar, reason: failed to generate SBOM: image scan error: scan error: scan failed: failed analysis: analyze error: pipeline error: failed to analyze layer (sha256:bff6683ea42c9216ad9427b6487fd00feadce5219a3a26e2923d563f6c4c865e): post analysis error: post analysis error: Unable to initialize the Java DB: Java DB update failed: DB download error: unacceptable media type: application/vnd.aquasec.trivy.db.layer.v1.tar+gzip" family=sbom
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0xe7115d]

goroutine 103 [running]:
github.com/openclarity/openclarity/scanner/families/sbom/types.(*Result).EncodeToBytes(0x0, {0xc00469a980?, 0x0?})
    /build/scanner/families/sbom/types/result.go:45 +0x7d
github.com/openclarity/openclarity/cli/presenter.(*DefaultPresenter).ExportSbomResult(0xc0061a2618, {0xc004984410?, 0xc003ccb080?}, {{0x65c8c31, 0x4}, {0x5bc2a20, 0x0}, {0x7a94000, 0xc0001e9860}})
    /build/cli/presenter/default.go:76 +0x8a
github.com/openclarity/openclarity/cli/presenter.(*DefaultPresenter).ExportFamilyResult(0xc0061a2600?, {0x7aee930?, 0xc00650ff50?}, {{0x65c8c31, 0x4}, {0x5bc2a20, 0x0}, {0x7a94000, 0xc0001e9860}})
    /build/cli/presenter/default.go:47 +0x8f
github.com/openclarity/openclarity/cli/presenter.(*MultiPresenter).ExportFamilyResult(0xc0001e9880?, {0x7aee930, 0xc00650ff50}, {{0x65c8c31, 0x4}, {0x5bc2a20, 0x0}, {0x7a94000, 0xc0001e9860}})
    /build/cli/presenter/multi.go:31 +0xd2
github.com/openclarity/openclarity/cli.(*CLI).FamilyFinished(0x0?, {0x7aee930?, 0xc00650ff50?}, {{0x65c8c31, 0x4}, {0x5bc2a20, 0x0}, {0x7a94000, 0xc0001e9860}})
    /build/cli/cli.go:65 +0x72
github.com/openclarity/openclarity/scanner/internal/family_runner.(*Runner[...]).Run(0x7a92660, {0x7aee968, 0xc005183b30}, {0x7aac308, 0xc0064ee300}, {0x7ad8998, 0xc007413760})
    /build/scanner/internal/family_runner/runner.go:79 +0x79a
github.com/openclarity/openclarity/scanner.New.newFamilyTask[...].func1({{0x7aac308, 0xc0064ee300}, {0x7ad8998, 0xc007413760}, 0xc00506aa80})
    /build/scanner/scanner.go:211 +0x12a
github.com/openclarity/openclarity/workflow/types.Task[...].Run(...)
    /build/workflow/types/task.go:36
github.com/openclarity/openclarity/workflow/internal/dispatcher.(*Dispatcher[...]).Dispatch.func1()
    /build/workflow/internal/dispatcher/dispatcher.go:121 +0xe8
created by github.com/openclarity/openclarity/workflow/internal/dispatcher.(*Dispatcher[...]).Dispatch in goroutine 100
    /build/workflow/internal/dispatcher/dispatcher.go:120 +0x48d
Stream closed EOF for openclarity/openclarity-scan-8ba61151-a678-46c2-88c4-21345e319c4e-kp7tv (openclarity-scan-8ba61151-a678-46c2-88c4-21345e319c4e)

```

## Affected Version

1.0.0

## Steps to Reproduce

1. Initiate a scan

## Checklist

<!-- TODO: Update the link below to point to your project's contributing guidelines -->
- [x] I have read the [contributing guidelines](/CONTRIBUTING.md)
- [x] I have verified this does not duplicate an existing issue

--- a/cli/presenter/writer_test.go
+++ b/cli/presenter/writer_test.go
@@ -1,0 +1,100 @@
+// Copyright © 2024 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package presenter
+
+import (
+	"os"
+	"reflect"
+	"testing"
+)
+
+func TestConsoleWriter(t *testing.T) {
+	writer := &ConsoleWriter{Output: os.Stdout}
+
+	tests := []struct {
+		name   string
+		b      []byte
+		prefix string
+		want   error
+	}{
+		{
+			name:   "nil bytes no error",
+			b:      nil,
+			prefix: "sbom",
+			want:   nil,
+		},
+		{
+			name:   "empty bytes no error",
+			b:      []byte{},
+			prefix: "sbom",
+			want:   nil,
+		},
+		{
+			name:   "test content no error",
+			b:      []byte("test"),
+			prefix: "sbom",
+			want:   nil,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := writer.Write(tt.b, tt.prefix); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("ConsoleWriter.Write() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestFileWriter(t *testing.T) {
+	writer := &FileWriter{Path: "."}
+
+	tests := []struct {
+		name     string
+		b        []byte
+		filename string
+		want     error
+	}{
+		{
+			name:     "nil bytes no error",
+			b:        nil,
+			filename: "sbom.cdx",
+			want:     nil,
+		},
+		{
+			name:     "empty bytes no error",
+			b:        []byte{},
+			filename: "sbom.cdx",
+			want:     nil,
+		},
+		{
+			name:     "test content no error",
+			b:        []byte("test"),
+			filename: "sbom.cdx",
+			want:     nil,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := writer.Write(tt.b, tt.filename); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("FileWriter.Write() = %v, want %v", got, tt.want)
+			}
+		})
+		// cleanup
+		os.Remove(tt.filename)
+	}
+}

--- a/scanner/families/sbom/types/result.go
+++ b/scanner/families/sbom/types/result.go
@@ -42,9 +42,12 @@ func (r *Result) EncodeToBytes(outputFormat string) ([]byte, error) {
 		return nil, fmt.Errorf("unable to parse output format: %w", err)
 	}
 
-	bomBytes, err := converter.CycloneDxToBytes(r.SBOM, f)
-	if err != nil {
-		return nil, fmt.Errorf("unable to encode results to bytes: %w", err)
+	bomBytes := []byte{}
+	if r.SBOM != nil {
+		bomBytes, err = converter.CycloneDxToBytes(r.SBOM, f)
+		if err != nil {
+			return nil, fmt.Errorf("unable to encode results to bytes: %w", err)
+		}
 	}
 
 	return bomBytes, nil

